libraries/AC_PrecLand/AC_PrecLand.cpp

```diff
@@ -602,12 +634,12 @@ bool AC_PrecLand::construct_pos_meas_using_rangefinder(float rangefinder_alt_m,
     Vector3f target_vec_unit_body;
     if (retrieve_los_meas(target_vec_unit_body)) {
         const struct inertial_data_frame_s *inertial_data_delayed = (*_inertial_history)[0];
-
+        const bool target_vec_valid = target_vec_unit_body.angle(_direction_of_approach) < radians(_cam_fov * 0.5f);
         const Vector3f target_vec_unit_ned = inertial_data_delayed->Tbn * target_vec_unit_body;
-        const bool target_vec_valid = target_vec_unit_ned.z > 0.0f;
         const bool alt_valid = (rangefinder_alt_valid && rangefinder_alt_m > 0.0f) || (_backend->distance_to_target() > 0.0f);
         if (target_vec_valid && alt_valid) {
-            float dist, alt;
+            // distance to target and distance to target along direction of approach
+            float dist, dist_along_doa;
```

Comment:
I wonder if we could improve these variables names.  "DOA" normally means dead-on-arrival so something more intuitive would be good.  Is "dist" perhaps the horizontal distance and dist_along_doa is the vertical distance (on a copter)?  .. or maybe one of them is 2D and one is 3D?  I'm not sure..

Reply:
Yes. I have changed variable name to "approach vector" aka _av_ in the code. 
The "dist" (I have changed it to "dist_to_target") is the length of the 3d vector to the target. And the "dist_along_doa" (now "dist_to_target_along_av") is the length of projection of target vector along approach vector, i.e., distance of copter from ground (if target is placed on ground).